engine/classes/Elgg/UploadService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Elgg;
+
+use Elgg\Http\Request;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * WARNING: API IN FLUX. DO NOT USE DIRECTLY.
+ *
+ * Use the elgg_* versions instead.
+ *
+ * @access private
+ * @since 2.3
+ */
+class UploadService {
+
+	/**
+	 * @var Request
+	 */
+	private $request;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Request $request Http request
+	 */
+	public function __construct(Request $request) {
+		$this->request = $request;
+	}
+
+	/**
+	 * Returns an array of uploaded file indices
+	 *
+	 * In case of a simple file input, will return [0]
+	 *
+	 * In case of a multiple file inputs with the same name, will produce ['pdf', 'doc']
+	 * <code>
+	 *   <input type="file" name="upload[pdf]" />
+	 *   <input type="file" name="upload[doc]" />
+	 * </code>
+	 *
+	 * In case of a file input with multiple attribute, will produce an array of numeric indices [0, 1, 2]
+	 * <code>
+	 *   <input type="file" name="upload[]" multiple />
+	 * </code>
+	 *
+	 * In case of a missing input element, will return an empty array
+	 *
+	 * @param string $input_name Form input name
+	 * @return array
+	 */
+	public function getUploadedFileIndices($input_name) {
```

Comment:
What's the use case for this function particularly? E.g. why not have it just return `UploadedFiles[]`?